### DISCUSSION TITLE
Create new state property for `alarm_control_panel` which is using an enum

### DISCUSSION THREAD

**Discussion by gjohansson-ST** [home-assistant/architecture#1140] [Entity Models] [answered]
## Background

Some entity models (like `alarm_control_panel`) allows integration to directly overwrite `state` property to set it's state using constants and are not modeled like the newer ones (as example `lawn_mower) which has a separate property for handling state using an enum.

## Proposal

Similar to the `lawn_mower` integration implement a separate state property `alarm_state` which is using an enum `AlarmControlPanelEntityState` for controlling it's state.

```python
@final
@cached_property
def state(self) -> str | None:
    """Return the current state."""
    return alarm_state

@cached_property
def alarm_state(self) -> AlarmControlPanelEntityState | None:
    """Return the current alarm control panel entity state.

    Integrations should overwrite this or use the 'attr_alarm_state'
    attribute to set the alarm status using the 'AlarmControlPanelEntityState' enum.
    """
    return self._attr_alarm_state
```

After the deprecation period we would change property to be `@final` and integrations should only implement `alarm_state` returning `AlarmControlPanelEntityState`.

Draft conceptual PR:https://github.com/home-assistant/core/pull/126283

## Update after initial proposal

- Return the enum member instead of stringifying it's state

**Comment by MartinHjelmare** [comment]
The `@final` decorator is only used by type checkers, right? It won't break any runtime code? Could we add it already at the start of the deprecation period (after all core integrations have been migrated)?

**Reply by gjohansson-ST** [reply]
We can do that. It's only type checking.
I intended to run a race to fix all core integrations hence in the proposal the deprecation notice should only be for custom components.

**Comment by MartinHjelmare** [comment, marked as answer]
We've discussed this proposal in the Core team and decided to approve it.

**Reply by frenck** [reply]
Implemented in https://github.com/home-assistant/developers.home-assistant/pull/2361
